sklearn/cluster/affinity_propagation_.py

```diff
@@ -111,8 +112,17 @@ def affinity_propagation(S, preference=None, convergence_iter=15, max_iter=200,
     if S.shape[0] != S.shape[1]:
         raise ValueError("S must be a square array (shape=%s)" % repr(S.shape))
 
-    if preference is None:
```

Comment:
This must continue to be supported, or its support deprecated over two versions, to ensure backwards compatibility